sql/mysqld.cc

```diff
@@ -1005,6 +1006,7 @@ static PSI_mutex_info all_server_mutexes[]=
   { &key_LOCK_gtid_waiting, "gtid_waiting::LOCK_gtid_waiting", 0},
   { &key_LOCK_thd_data, "THD::LOCK_thd_data", 0},
   { &key_LOCK_thd_kill, "THD::LOCK_thd_kill", 0},
+  { &key_LOCK_temporary_tables, "THD::LOCK_temporary_tables", 0},
```

Comment:
Are we going to implement a LOCK for every member of THD struct, going forward? Why LOCK_thd_data is not sufficient? What happens if you just use that one?

Reply:
good to know, thanks!

Reply:
Are there any updates on this issue?